packages/hash/api/src/graphql/resolvers/knowledge/entity/before-update-entity-hooks.ts

```diff
@@ -77,6 +77,17 @@ const userEntityHookCallback: BeforeUpdateEntityHookCallback = async ({
       throw new ApolloError("Cannot unset preferred name");
     }
   }
+
+  const currentEmails = userModel.getEmails();
+
+  const updatedEmails: string[] =
+    updatedProperties[SYSTEM_TYPES.propertyType.email.baseUri];
+
+  if (currentEmails.sort().join() !== updatedEmails.sort().join()) {
+    await userModel.updateKratosIdentityTraits({
+      emails: updatedEmails,
+    });
+  }
```

Comment:
This will detect a difference if different casing has been used for the same email address. This is only important if this triggers further side effects like validation emails (which are unnecessary if the address hasn't actually changed)

Reply:
That's a good spot, one way of not having to account for this would be to have an `EmailAddress` data-type that constrains that an email address can only be composed of lower-case letters.

cc @Alfred-Mountfield @thehabbos007

Reply:
We could constrain it, but generally the email spec just ignores casing (to my knowledge) so we could do the same, as in, just ignore it.

Reply:
That's interesting, either we enforce one casing in the Graph Layer or we ignore case when comparing (I believe emails are case insensitive). I don't know which would make most sense, but I think we could change the comparison here to be case-insensitive.

Reply:
> think we could change the comparison here to be case-insensitive.

Agreed, I think we can (and maybe should) do this regardless of if we enforce everything is lower-case later on

Reply:
Makes sense, have addressed in https://github.com/hashintel/hash/pull/1364/commits/1ba348765e96e3e1d4f724069618e6b42a3ef482